packages/kbn-config-schema/src/types/intersection_type.test.ts

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { schema } from '../..';
+
+describe('schema.allOf', () => {
```

Comment:
Might be worth adding/updating a test to ensure the schema type return is what we expect, i.e.:

```ts
const obj: Type<A & B> = schema.allOf(...)
```

WDYT?

Reply:
Yeah, makes sense. We don't really have such "type tests" for other APIs of config-schema, and we probably should